scripts/morty/analysis.py

```diff
@@ -0,0 +1,273 @@
+        ----------
+        result_mat : :class:`np.ndarray`
+            Holds values over time, space, and nuclide (in that order)
+
+        Returns
+        -------
+        parasitic_abs_val : :class:`np.ndarray`
+            Parasitic absorption rates for nuclides over time in
+            atoms per cc
+        """
+        parasitic_abs_val = []
+        for nuclide in range(self.run_params['num_nuclides']):
+            paras_rate = []
+            for ti, t in enumerate(self.run_params['times']):
+                spat_avg_conc = np.mean(result_mat[ti, :, nuclide])
+                parasitic = spat_avg_conc * \
+                    self.data_params['loss_rates'][nuclide]
+                paras_rate.append(parasitic)
+            integral_form = integrate.cumtrapz(
+                paras_rate, self.run_params['times'])
+            integral_form = np.insert(integral_form, 0, 0.0)
+            parasitic_abs_val.append(integral_form)
+        return parasitic_abs_val
+
+    def ode_pde_compare(self):
+        """
+        Run with current parameters for both ODE and PDE solver methods.
+
+        Returns
+        -------
+        data : dict
+            key : str
+                Name of variable
+        """
+        current_solver = self.run_params['solver_method']
+        methods = ['ODE', 'PDE']
+        ylab = 'Concentration [atoms/cc]'
+        savename = 'PDE_ODE_comparison'
+        time_factor, xlab = self._time_lab()
+        xs, ys, labs = self._method_change(methods, 'solver_method',
+                                           time_factor)
+        self._save_data(xs, ys, labs, xlab, ylab, savename)
+
+        self.run_params['solver_methods'] = current_solver
+        data = self.data
+        return data
+
+    def nuclide_refinement(self, max_nuc=5):
+        """
+        Run with current parameters for varying number of nuclides.
+
+        Parameters
+        ----------
+        max_nuc : int
+            Maximum number of nuclides
+
+        Returns
+        -------
+        data : dict
+            key : str
+                Name of variable
+        """
+        current_nuclides = self.run_params['num_nuclides']
+        methods = np.linspace(1, max_nuc, max_nuc).astype(int)
+        ylab = 'Concentration [atoms/cc]'
+        savename = 'nuclide_refinement'
+        time_factor, xlab = self._time_lab()
+        xs, ys, labs = self._method_change(methods, 'num_nuclides',
+                                           time_factor,
+                                           methodname_extension=' nuclides')
+        self._save_data(xs, ys, labs, xlab, ylab, savename)
+
+        self.run_params['num_nuclides'] = current_nuclides
+        data = self.data
+        return data
+
+    def spatial_refinement(self, spat_node_list=[10, 100, 1000, 10000]):
+        """
+        Run with current parameters for varying number of spatial nodes.
+
+        Parameters
+        ----------
+        spat_node_list : list of int
```

Comment:
The extra two letters in `spatial_node_list` might be helpful . . . Since you know its a list by asking its type, you can even shorten the name by calling it `spatial_nodes`.